services/load/service.go

```diff
@@ -0,0 +1,416 @@
+package load
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/ghodss/yaml"
+
+	"github.com/influxdata/kapacitor/client/v1"
+	"github.com/pkg/errors"
+)
+
+var defaultURL = "http://localhost:9092"
+
+type Service struct {
+	mu     sync.Mutex
+	config Config
+
+	cli    *client.Client
+	logger *log.Logger
+}
+
+func NewService(c Config, l *log.Logger) (*Service, error) {
+	cli, err := client.New(client.Config{
+		URL: defaultURL,
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to create client: %v", err)
+	}
+
+	return &Service{
+		config: c,
+		logger: l,
+		cli:    cli,
+	}, nil
+}
+
+// TaskFiles gets a slice of all files with the .tick file extension
+// in the configured task directory.
+func (s *Service) taskFiles() (tickscripts []string, err error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	tasksDir := s.config.tasksDir()
+
+	files, err := ioutil.ReadDir(tasksDir)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, file := range files {
+		if file.IsDir() {
+			continue
+		}
+
+		filename := file.Name()
+		switch ext := filepath.Ext(filename); ext {
+		case ".tick":
+			tickscripts = append(tickscripts, filepath.Join(tasksDir, filename))
+		default:
+			continue
+		}
+	}
+
+	return
+}
+
+// TemplateFiles gets a slice of all files with the .tick file extension
+// and any associated files with .json, .yml, and .yaml file extentions
+// in the configured template directory.
+func (s *Service) templateFiles() (tickscripts []string, tmplVars []string, err error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	templatesDir := s.config.templatesDir()
+
+	files, err := ioutil.ReadDir(templatesDir)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	for _, file := range files {
+		if file.IsDir() {
+			continue
+		}
+
+		filename := file.Name()
+		switch ext := filepath.Ext(filename); ext {
+		case ".tick":
+			tickscripts = append(tickscripts, filepath.Join(templatesDir, filename))
+		case ".yml", ".json", ".yaml":
+			tmplVars = append(tmplVars, filepath.Join(templatesDir, filename))
+		default:
+			continue
+		}
+	}
+
+	return
+}
+
+// HandlerFiles gets a slice of all files with the .json, .yml, and
+// .yaml file extentions in the configured handler directory.
+func (s *Service) handlerFiles() ([]string, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	handlers := []string{}
+
+	handlersDir := s.config.handlersDir()
+
+	files, err := ioutil.ReadDir(handlersDir)
```

Comment:
in `load/config.go` the `Validate` method requires that the directory specified must have subdirectories `tasks`, `templates`, and `handlers`. Should I relax this requirement?